### ISSUE TITLE
Copied images are not saved to disk (Windows 10)

### ISSUE THREAD

**Issue by pgunnink** [typora/typora-issues#1626]
On Windows 10, when you past an image into a file, the image is saved as base64, even though I checked the setting 'Copy image to ./assets'. Also, nothing appears in ./assets. When manually trying to copy the image by right clicking and selecting Copy image to... I get the error "Failed to copy image to target folder. TypeError: Path must be a string. Received undefined.

**Comment by abnerlee** [comment]
Does your md file saved in disk? `./assets` is relative to your md file, so the md file must have a path

**Comment by pgunnink** [comment]
Yeah it is, I can see `./assets` being created at the proper place in path, the image is just not saved.

**Comment by abnerlee** [comment]
So from where/how did you copy image? Could you provide sample URL if it is from webpage?

**Comment by pgunnink** [comment]
I copied it from a local webpage, so no URL. Images that I have already saved locally work fine, they show in markdown as `![1531925578675](local path)`. When I check the setting save to `./assets` the image also gets copied to `./assets`.

**Comment by abnerlee** [comment]
Thanks for the info.

Is it possible to send us <hi@typora.io> a sample local HTML file for us to check this issue?

**Comment by pgunnink** [comment]
It also happens when I try to copy images from Google Colab, an online version of Jupyter, so you could try it from there. This [link](https://colab.research.google.com/notebooks/welcome.ipynb#scrollTo=FlQq0SUepQbd) should take to a sample page with an image, you could try that.

**Comment by xingdaos** [comment]
I have happending too，when I copy a screenshot to the md document .
and I have set:
![image](https://user-images.githubusercontent.com/33388842/44780328-4a12a900-abb4-11e8-9b82-c36c2e9495d3.png)
but the path like:
![image](https://user-images.githubusercontent.com/33388842/44780443-8ba35400-abb4-11e8-9cc0-242a3c8c40cd.png)
but it is likly only happed before the file have not to saved , when it is saved , it ok .
but the photos before to saved the md document , it is not auto save to .asset directory , too danger .

**Comment by nunhes** [comment]
I'm sorry. I found this problem since few days ago in version 1.9.2-dev. The program is configured to copy images to the ./assets/ folder that exists in my project folder, but... the images are not saved, the icon or loader appears on the image but the task never finishes...
